### ISSUE TITLE
Make wheels for python 3.11

### ISSUE THREAD

**Issue by vient** [tensorflow/io#1730]
It is impossible to install tensorflow-io or tensorflow-io-gcs-filesystem via pip at the moment: pypi does not have 3.11 wheels, and `python -m pip install git+https://github.com/tensorflow/io.git@v0.27.0 --global-option=--project --global-option=tensorflow-io-gcs-filesystem` fails because pip does not pass custom options to `setup.py egg_info` so it fails with unmet dependency (note that I tried to install tensorflow-io-gcs-filesystem by passing --project option but pip reports unmet dependency for tensorflow-io):
```
ERROR: Could not find a version that satisfies the requirement tensorflow-io-gcs-filesystem==0.27.0 (from tensorflow-io) (from versions: none)
ERROR: No matching distribution found for tensorflow-io-gcs-filesystem==0.27.0
```

In other words, problem with pip + git method is that tensorflow-io and tensorflow-io-gcs-filesystem are built from the same setup.py, pip has pretty limited control there.

**Comment by vient** [comment]
This is blocker for installing tensorflow itself to 3.11, for example, using `python -m pip install git+https://github.com/tensorflow/tensorflow.git@167c9eb7cc006f519dc731f27a7c9df287225d40#subdirectory=tensorflow/tools/pip_package`.

**Comment by vient** [comment]
Will be solved by #1740
